src/serviceDesks/incontact/inContactServiceDesk.ts

```diff
@@ -133,21 +134,26 @@ class InContactServiceDesk implements ServiceDesk {
         if (output.status) {
           switch (output.status) {
             case 'Active':
+              // For production, should change this to load in actual agent profile info
               this.callback.agentJoined(this.agent);
               break;
             case 'Waiting':
               try {
+                // eslint-disable-next-line no-await-in-loop
                 const request = await fetch(`${this.SERVER_BASE_URL}/incontact/queue`, {
                   method: 'POST',
                   headers: { Accept: 'application/json', 'Content-Type': 'application/json' },
-                  body: JSON.stringify({ ...this.session }),
+                  body: JSON.stringify(this.session),
                 });
+                // eslint-disable-next-line no-await-in-loop
                 const output = await request.json();
-                if (output && output.queue != undefined) {
-                  this.callback.updateAgentAvailability({ position_in_queue: output.queue });
+                if (output.queue !== undefined) {
+                  // Add one since queue value is 0 when there is no line
+                  this.callback.updateAgentAvailability({ position_in_queue: output.queue + 1 });
                 }
               } catch (error) {
-                return Promise.reject();
+                // Do not stop polling when queue call fails
+                console.error('Unable to retrieve queue information.');
```

Comment:
You are committing the error here of catching an error and then throwing it away. You really pretty much never want to do that since you're throwing away potentially useful information like what the error actually was or the stack trace. In this case just add `, error` to the console output so you log the error as well.